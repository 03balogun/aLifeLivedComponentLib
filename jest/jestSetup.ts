/* eslint-disable no-undef */
jest.useFakeTimers('legacy'); // Stops reference error from appearing
