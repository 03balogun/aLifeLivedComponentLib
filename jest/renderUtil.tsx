import React from 'react';
import { ThemeProvider } from '../src/themes/ThemeProvider';
import { render as rtlRender } from '@testing-library/react-native';

function render(ui, { ...renderOptions } = {}) {
  function Wrapper({ children }) {
    return <ThemeProvider>{children}</ThemeProvider>;
  }

  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

// re-export everything for convenience
export * from '@testing-library/react-native';

// override render method
export { render };
