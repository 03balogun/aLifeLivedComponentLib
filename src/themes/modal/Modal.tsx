import React from 'react';
import { TouchableWithoutFeedback } from 'react-native';
// Components
import { Box } from '../Box';

type Props = {
  children: React.ReactNode;
  onPress: () => void;
};

export const Modal = ({ onPress, children }: Props) => {
  return (
    <TouchableWithoutFeedback onPress={() => onPress()}>
      <Box flex={1} width="100%" justifyContent="center">
        <Box backgroundColor="primaryHeavy" margin="s" borderRadius="medium">
          {children}
        </Box>
      </Box>
    </TouchableWithoutFeedback>
  );
};
