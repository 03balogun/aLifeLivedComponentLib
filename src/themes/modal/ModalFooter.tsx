import React from 'react';
import { Box } from '../Box';

type ModalHeaderProps = {
  children: React.ReactNode;
};

export const ModalFooter: React.FC<ModalHeaderProps> = props => {
  return (
    <Box margin="m" flexDirection="row" alignSelf="center" flexWrap="wrap">
      {props.children}
    </Box>
  );
};
