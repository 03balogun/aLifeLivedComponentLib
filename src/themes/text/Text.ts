import { createText } from '@shopify/restyle';
import { Theme } from '../theme';

export const textVariants = {
  h1: {
    fontSize: 32,
    lineHeight: 38,
    color: 'contrastHighest',
  },
  h2: {
    fontSize: 24,
    lineHeight: 30,
    color: 'contrastHighest',
  },
  h3: {
    fontSize: 16,
    lineHeight: 24,
    color: 'contrastHighest',
  },
  body: {
    fontSize: 16,
    lineHeight: 20,
    color: 'contrastHighest',
  },
  caption: {
    fontSize: 14,
    lineHeight: 18,
    color: 'contrastLow',
  },
  default: {
    fontSize: 16,
    lineHeight: 20,
    color: 'contrastHighest',
  },
};

export const Text = createText<Theme>();
