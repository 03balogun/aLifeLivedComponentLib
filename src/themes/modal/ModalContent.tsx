import React from 'react';
import { Box } from '../Box';
import { Text } from '../text/Text';

type ModalHeaderProps = {
  header: string;
  description?: string;
};

export const ModalContent: React.FC<ModalHeaderProps> = ({
  header,
  description,
}) => {
  return (
    <Box m="m" paddingBottom="s" alignContent="center" alignItems="center">
      <Text variant="h2" fontWeight="bold">
        {header}
      </Text>
      {description && (
        <Text variant="body" marginTop="m">
          {description}
        </Text>
      )}
    </Box>
  );
};
