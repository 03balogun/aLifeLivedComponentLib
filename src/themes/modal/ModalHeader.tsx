import React from 'react';
import { Box } from '../Box';
import { SystemPressable } from '../systemPressable/SystemPressable';
import { Text } from '../text/Text';

type ModalHeaderProps = {
  onHeaderPress: () => void;
};

export const ModalHeader: React.FC<ModalHeaderProps> = ({ onHeaderPress }) => {
  return (
    <Box m="s" alignContent="flex-start" alignItems="flex-start">
      <SystemPressable onPress={() => onHeaderPress()}>
        <Text variant="h3" fontWeight="bold">
          X
        </Text>
      </SystemPressable>
    </Box>
  );
};
