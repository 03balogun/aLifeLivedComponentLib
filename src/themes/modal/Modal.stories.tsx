import React from 'react';
import { storiesOf } from '@storybook/react-native';
import { action } from '@storybook/addon-actions';

// Component
import { Modal } from './Modal';
import { ModalHeader } from './ModalHeader';
import { ModalFooter } from './ModalFooter';
import { Button } from '../../components/button/Button';
import { ModalContent } from './ModalContent';

storiesOf('Modal', module)
  .addDecorator(Story => Story())
  .add('Modal with header only', () => (
    <Modal onPress={action('Modal action fired')}>
      <ModalHeader header="This is testing the modal" />
      <ModalContent header="Test content" />
      <ModalFooter>
        <Button
          title="Test button one"
          onPress={action('Fire on press for button one')}
          width="50%"
          backgroundColor="primary"
          color="primary"
          marginRight="s"
        />
        <Button
          title="Test button two"
          onPress={action('Fire on press for button one')}
          width="50%"
          marginLeft="s"
        />
      </ModalFooter>
    </Modal>
  ));
