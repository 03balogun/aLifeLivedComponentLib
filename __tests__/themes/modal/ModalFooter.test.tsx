import React from 'react';
import { render } from '../../../jest/renderUtil';

// Components
import { ModalFooter } from '../../../src/themes/modal/ModalFooter';
import { Text } from '../../../src/themes/text/Text';

const setup = () => {
  const utils = render(
    <ModalFooter>
      <Text>Modal Footer!</Text>
    </ModalFooter>
  );

  return {
    ...utils,
  };
};

describe('themes >> ModalFooter', () => {
  describe('Render modal footer', () => {
    test('Should render modal footer correctly with children', () => {
      const { toJSON } = setup();
      expect(toJSON()).toMatchSnapshot();
    });

    test('Should find children', () => {
      const { getByText } = setup();
      expect(getByText('Modal Footer!')).not.toBeNull();
    });
  });
});
