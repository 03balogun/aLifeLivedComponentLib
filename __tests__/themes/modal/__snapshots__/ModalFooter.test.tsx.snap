// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`themes >> ModalFooter Render modal footer Should render modal footer correctly with children 1`] = `
<View
  style={
    Array [
      Object {
        "alignSelf": "center",
        "display": "flex",
        "flexDirection": "row",
        "margin": 16,
      },
      Array [
        Object {},
      ],
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "display": "flex",
        },
      ]
    }
  >
    Modal Footer!
  </Text>
</View>
`;
